Firestore/core/src/firebase/firestore/remote/grpc_stream.cc

```diff
@@ -226,30 +246,35 @@ void GrpcStream::FastFinishCompletionsBlocking() {
 
 bool GrpcStream::WriteAndFinish(grpc::ByteBuffer&& message) {
   bool did_last_write = false;
+  if (!is_grpc_call_finished_) {
+    did_last_write = TryLastWrite(std::move(message));
```

Comment:
This refactoring is to make the handling of `is_grpc_call_finished_` in this case more explicit.